### PR TITLE
Component refactor connect to #209

### PR DESCRIPTION
**Pull request by Casey1449**
- Creates fullscreen loader component.
- Creates Campaign flow refactor with subroutes! Nifty green dots!

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,2 @@
+{
```
Comment (wdoug) [inline .vscode/settings.json]: This seems unnecessary. I wonder if we should gitignore the whole `.vscode` directory.

File: client/easy_to_forget_to_dos.txt [hunk 2]
```diff
@@ -0,0 +1,2 @@
+- initial input field changes on focus: blue haze, or transform/grow like material gets closer.
+- put 'chosen address' in redux
```
Comment (wdoug) [inline client/easy_to_forget_to_dos.txt]: Are these todos supposed to be done in this PR?

File: client/src/containers/ChooseCampaign.js [hunk 3]
```diff
@@ -1,171 +1,193 @@
...
-  constructor(props) {
-    super(props);
-    this.state = {
-      selectedOption: null
-    };
-    this.handleOptionChange = this.handleOptionChange.bind(this);
-    this.handleOptionChange = this.handleOptionChange.bind(this);
-    this.handleFormSubmit = this.handleFormSubmit.bind(this);
-  }
+	constructor(props) {
```
Comment (wdoug) [inline client/src/containers/ChooseCampaign.js]: Spaces -> tabs = less than ideal

Comment (Casey1449) [reply client/src/containers/ChooseCampaign.js]: Just switched in in my workspace settings for this project, so I imagine it'll get corrected from here on out as we go. But you'll never convert me on an ideological level!

File: client/easy_to_forget_to_dos.txt (hunk 2 above)
Comment (Casey1449) [reply client/easy_to_forget_to_dos.txt]: Nah, this file is super old, from before we had waffle I think. I just deleted it.

File: .vscode/settings.json (hunk 1 above)
Comment (Casey1449) [reply .vscode/settings.json]: done.

File: client/src/redux/actions/initialSearch.js [hunk 4]
```diff
@@ -62,34 +62,35 @@ export function searchAddressFlow(address, latLngHelper) {
...
-  };
-}
+
+// export function createCampaignFailure(error) {
+//   return {
+//     type: 'CREATE_CAMPAIGN_FAILURE',
+//     error
+//   };
+// }
+
```
Comment (ximsce) [inline client/src/redux/actions/initialSearch.js]: are you leaving these commented code blocks in for any particular purpose? If not, please remove them. Thanks!

File: client/src/redux/actions/newCampaign.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+// 	return async dispatch => {
+// 		try {
+// 			const { response } = await dispatch(createCampaign(campaignInfo));
+// 			browserHistory.push(`/campaign/${response.data._id}`);
+// 		} catch (err) {
+// 			console.log(err);
+// 		}
+// 	};
+// }
+
```
Comment (ximsce) [inline client/src/redux/actions/newCampaign.js]: same here, please remove commented out code blocks or provide a note about what you are trying to save here

**Pull request merged**